UnityProject/Assets/Scripts/UI/Systems/MainHUD/PacketLossUI.cs

```diff
@@ -0,0 +1,96 @@
+﻿using System;
+using System.Linq;
+using Mirror;
+using UnityEngine;
+using UnityEngine.UI;
+using WebSocketSharp;
+using System.Net.NetworkInformation;
+using System.Threading;
+using Ping = System.Net.NetworkInformation.Ping;
+
+namespace UI.Systems.MainHUD
+{
+	public class PacketLossUI : MonoBehaviour
+	{
+		[SerializeField] private Text packetLossText;
+
+		private CancellationToken token;
+		private double packetLossPercentage = 0;
+		private readonly int pingCount = 10;
+		private int pingNumber = 0;
+		private bool canPingAgain = true;
+		private DateTime startTime;
+		private readonly Ping ping = new Ping();
+
+		private int[] attempts = new[] { 0, 0, 0, 0, 0, 0, 0, 0, 0 , 0 };
+
+		private void Awake()
+		{
+			CancellationTokenSource source = new CancellationTokenSource();
+			token = source.Token;
+			ping.PingCompleted += Loss;
+		}
+
+		private void Update()
+		{
+			MeasurePacketLoss();
```

Comment:
surely there must be some inbuilt methods inside of mirror

Reply:
There isn't